lang/LangPrimSource/PyrFilePrim.cpp

```diff
@@ -1326,6 +1326,67 @@ int prPipeOpen(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+int prPipeOpenArgv(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *callerSlot, *argsSlot, *modeSlot;
+	char mode[12];
+
+	callerSlot = g->sp - 2;
+	argsSlot = g->sp - 1;
+	modeSlot = g->sp;
+
+#ifdef SC_IPHONE
+	SetInt(callerSlot, 0);
+	return errNone;
+#endif
+
+	if (NotObj(argsSlot)) return errWrongType;
+
+	if (NotObj(modeSlot) || !isKindOf(slotRawObject(modeSlot), class_string))
+		return errWrongType;
+	if (slotRawObject(modeSlot)->size > 11) return errFailed;
+
+	PyrFile *pfile = reinterpret_cast<PyrFile*>(slotRawObject(callerSlot));
+
+	PyrObject *argsColl = slotRawObject(argsSlot);
+	if (!(slotRawInt(&argsColl->classptr->classFlags) & classHasIndexableInstances))
+		return errNotAnIndexableObject;
+
+	if( argsColl->size < 1)
+		return errFailed;
```

Comment:
i'd appreciate it if some of these sections had explicit comments about what's being checked; my eyes glaze over when i try to read all these type checks (which are necessary, just a lot of them..)